### PR TITLE
several bug fixes to client

### PR DESCRIPTION
**Pull request by daringordon**
see: https://stripe.com/docs/api/charges/object#charge_object-payment_method_details-card-brand

### CODE REVIEW

**Comment by daringordon** [comment]
1) detach payment method
2) create setupintent enhancement 
3) payment_intent update enhancement
4) aligned the expected serde values for CardBrand with those received from the stripe api
5) add transfer_data to payment_intent update

File: src/resources/setup_intent.rs [hunk 1]
```diff
@@ -266,23 +266,20 @@ pub struct CreateSetupIntent<'a> {
...
+    ///
+    /// Use `off_session` if your customer may or may not be in your checkout flow.
+    /// If not provided, this value defaults to `off_session`.
+    pub usage: String
 }
 
 impl<'a> CreateSetupIntent<'a> {
     pub fn new() -> Self {
         CreateSetupIntent {
-            confirm: Default::default(),
```
Comment (stearnsc) [inline src/resources/setup_intent.rs:274]: Should this fn just return Default::default()?

File: src/resources/card.rs [hunk 2]
```diff
@@ -154,26 +154,26 @@ pub enum CheckResult {
 
 #[derive(Copy, Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
 pub enum CardBrand {
-    #[serde(rename = "American Express")]
```
Comment (stearnsc) [inline src/resources/card.rs:157]: I don't think this is safe - 
https://stripe.com/docs/api/cards/object#card_object-brand

Would this meet your needs? https://github.com/wyyerd/stripe-rs/pull/188

**Review by stearnsc — CHANGES_REQUESTED** [review]

**Comment by stearnsc** [comment]
I think this looks good other than the CardBrand issue - if https://github.com/wyyerd/stripe-rs/pull/188 meets your needs for that portion, I'm happy merging the other parts.

**Comment by seanpianka** [comment]
Thanks for the review, @stearnsc! 

@daringordon, would you be interested in submitting separate PRs for these different changes? I believe the first change has now been covered by #141.

File: src/resources/card.rs (hunk 2 above)
Comment (seanpianka) [reply src/resources/card.rs:157]: #188 has now been merged. The `enum PaymentMethodCardBrand` now holds variants that would be helpful here.

Comment (daringordon) [reply src/resources/card.rs:157]: Yes, #188 covers.  Thanks!

**Pull request closed without merging**